engine/api/workflow.go

```diff
@@ -213,8 +212,8 @@ func (api *API) putWorkflowHandler() Handler {
 			return sdk.WrapError(errHr, "putWorkflowHandler> HookRegistration")
 		}
 
-		if wf.Root.Context.DefaultPayload != nil || (wf.Root.Context != nil && wf.Root.Context.Application != nil && wf.Root.Context.Application.RepositoryFullname != "") {
-			wf.Metadata = getUpdatedMetadata(wf.Metadata)
+		if defaultTags, ok := wf.Metadata["default_tags"]; wf.Root.IsLinkedToRepo() && (!ok || defaultTags == "") {
+			wf.Metadata["default_tags"] = "git.branch,git.author"
```

Comment:
log.go:149: [ERROR]    [PANIC_RECOVERY] Panic occurred on PUT:/Test_putWorkflowHandler/project/3zmjcqyqlu/workflows/Name, recover assignment to entry in nil map

			/go/src/github.com/ovh/cds/engine/api/workflow.go:216 +0x776